roles/insights/tasks/main.yml

```diff
@@ -0,0 +1,16 @@
+---
+
+- name: internal_host_started_processing_role
+  set_fact:
+    internal_host_started_processing_role: "insights"
+
+- name: gather insights.machine-id
+  raw: if [ -f /etc/insights-client/machine-id ] ; then cat /etc/insights-client/machine-id; fi || if [ -f /etc/redhat-access-insights ] ; then cat /etc/redhat-access-insights/machine-id; fi
+  register: internal_cat_insights_client_id
+  ignore_errors: yes
+
+- name: set instnum
+  set_fact:
+    insights_client_id: "{{ internal_cat_insights_client_id.get('stdout') | default('') }}"
+  ignore_errors: yes
+  when: '"stdout" in internal_cat_insights_client_id'
```

Comment:
you can choose if you want this when (with no default) or if you want to just have the default, they will both do the same thing.